app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.updateRegisteredDetails.Submission
+
+import models.{RetrievedSubscription, UserAnswers}
+import pages.updateRegisteredDetails.UpdateContactDetailsPage
+import play.api.libs.json.{Json, Writes}
+
+case class VariationsPersonalDetails(
+                                      name: Option[String] = None,
+                                      position: Option[String] = None,
+                                      telephoneNumber: Option[String] = None,
+                                      emailAddress: Option[String] = None) {
+  def nonEmpty: Boolean = Seq(name, position, telephoneNumber, emailAddress).flatten.nonEmpty
+}
+
+object VariationsPersonalDetails {
+
+  def apply(userAnswers : UserAnswers, subscription : RetrievedSubscription):Option[VariationsPersonalDetails] = {
+    userAnswers.get(UpdateContactDetailsPage).flatMap{ answers =>
+      val name = if(answers.fullName == subscription.contact.name.getOrElse(None)){None} else {Some(answers.fullName)}
+      val position = if(answers.position == subscription.contact.positionInCompany.getOrElse(None)){None} else {Some(answers.position)}
```

Comment:
should these be 
subscription.contact.name.getOrElse("")
subscription.contact.positionInCompany.getOrElse("")
i don't think answers.fullname and answers.position are optional so its comparing unrelated types?

Reply:
in retrieved subscription these are optional

Reply:
and optional in VariationsPersonalDetails

Reply:
not optional in user answers though
userAnswers.fullName (this is mandatory) comparing it to the subscription get or else none
so potentially youre comparing user answers.fullname to None,
it should be userAnswers.fullname compared to subscription answer OR ""